stackslib/src/chainstate/stacks/boot/pox_4_tests.rs

```diff
@@ -3282,6 +3283,323 @@ fn stack_agg_commit_verify_sig() {
+        bob.nonce,
+        carl.principal.clone(),
+        amount,
+        bob.pox_address.clone(),
+        burn_block_height as u128,
+        lock_period,
+    );
+    bob.nonce += 1;
+    // Bob pool operator calling delegate-stack-stx on behalf of Dave
+    let bob_delegate_stack_stx_for_dave_tx = make_pox_4_delegate_stack_stx(
+        &bob.private_key,
+        bob.nonce,
+        dave.principal.clone(),
+        amount,
+        bob.pox_address.clone(),
+        burn_block_height as u128,
+        lock_period,
+    );
+    bob.nonce += 1;
+
+    // Aggregate Commit
+    let bobs_aggregate_commit_index_tx = make_pox_4_aggregation_commit_indexed(
+        &bob.private_key,
+        bob.nonce,
+        &bob.pox_address,
+        next_reward_cycle,
+        Some(alice_signature_initial),
+        &alice.public_key,
+        u128::MAX,
+        1,
+    );
+    bob.nonce += 1;
+
+    let txs = vec![
+        carl_delegate_stx_to_bob_tx.clone(),
+        dave_delegate_stx_to_bob_tx.clone(),
+        bob_delegate_stack_stx_for_carl_tx.clone(),
+        bob_delegate_stack_stx_for_dave_tx.clone(),
+        bobs_aggregate_commit_index_tx.clone(),
+    ];
+
+    // Advance to next block in order to collect aggregate commit reward index
+    target_height += 1;
+    let (latest_block, tx_block) = advance_to_block_height(
+        &mut peer,
+        &observer,
+        &txs,
+        &mut peer_nonce,
+        target_height.into(),
+    );
+
+    // Get Bob's aggregate commit reward index
+    let bob_aggregate_commit_reward_index_actual = &tx_block
+        .receipts
+        .get(5)
+        .unwrap()
+        .result
+        .clone()
+        .expect_result_ok()
+        .unwrap();
+    let bob_aggregate_commit_reward_index_expected = Value::UInt(0);
+    assert_eq!(
+        bob_aggregate_commit_reward_index_actual,
+        &bob_aggregate_commit_reward_index_expected
+    );
+
+    // Eve Late Functions
+    // Eve pool stacker late delegating STX to Bob
+    let eve_delegate_stx_to_bob_tx = make_pox_4_delegate_stx(
+        &eve.private_key,
+        eve.nonce,
+        amount,
+        bob.principal.clone(),
+        None,
+        Some(bob.pox_address.clone()),
+    );
+    eve.nonce += 1;
+    // Bob pool operator calling delegate-stack-stx on behalf of Eve
+    let bob_delegate_stack_stx_for_eve_tx = make_pox_4_delegate_stack_stx(
+        &bob.private_key,
+        bob.nonce,
+        eve.principal.clone(),
+        amount,
+        bob.pox_address.clone(),
+        burn_block_height as u128,
+        lock_period,
+    );
+    bob.nonce += 1;
+    // Bob's Aggregate Increase
+    let bobs_aggregate_increase = make_pox_4_aggregation_increase(
+        &bob.private_key,
+        bob.nonce,
+        &bob.pox_address,
+        next_reward_cycle,
+        bob_aggregate_commit_reward_index_actual
+            .clone()
+            .expect_u128()
+            .unwrap(),
+        Some(alice_signature_increase),
+        &alice.public_key,
```

Comment:
You should add a check here to verify that passing a different signing key other than what was given in the initial `stack-aggregation-commit` will fail.

Reply:
Test now included.